### DISCUSSION TITLE
How to get job status?

### DISCUSSION THREAD

**Discussion by tadejsv** [riverqueue/river#188] [Q&A] [answered]
I can't find any way of getting the status of a submitted job. 

The only way would be to use Subscribe - but this would involve a lot more work (going through all events of all jobs and filtering down to only those for the job I am interested in).

This functionality is needed when the caller needs to wait on the job to finish.

**Comment by brandur** [comment, marked as answer]
A "job get" API came in with #186 and should be available in a release soon.

**Reply by dhermes** [reply]
Amazing! I was just about to ask the same question. Happy to see there is an answer already.
